pkg/driver/common/common.go

```diff
@@ -71,6 +74,44 @@ type Zcerts struct {
 	Certs []*certs.ZCert `json:"certs,omitempty"` // EVE device certs
 }
 
+//ApiRequest stores information about requests from EVE
+type ApiRequest struct {
+	Timestamp time.Time `json:"timestamp"`
+	UUID      uuid.UUID `json:"uuid,omitempty"`
+	ClientIP  string    `json:"client-ip"`
+	Forwarded string    `json:"forwarded,omitempty"`
+	Method    string    `json:"method"`
+	URL       string    `json:"url"`
+}
+
+//PCRValue stores one single PCR value from TPM, from a particular hash bank
+type PCRValue struct {
+	Index uint32 `json:"index"`
+	Value string `json:"value"` // may contain '*' to allow any value in template
+}
+
+//PCRTemplate stores template with EVE version, Firmware version, GPSInfo and set of PCRValues
+type PCRTemplate struct {
+	EveVersion      string                       `json:"eveVersion"`
+	FirmwareVersion string                       `json:"firmwareVersion"`
+	PCRValues       []*PCRValue                  `json:"PCRValues"`
+	GPSInfo         *attest.AttestGPSCoordinates `json:"GPSInfo"`
```

Comment:
Having the raw GPS values in here isn't going to be useful since one would need some geofencing (defining a min,max lat and long in the template) to make any useful comparisons.
So should at least document that GPSInfo needs to change to be useful. Or leave it out for now. Note that EVE-OS does not currently send any GPS coordinates in the attestation message; code has recently been added to send it as part of a location info message, but since it takes minutes for a GPS modem to get reliable GPS coordinates on power on it is hard to send something useful on power on. @milan-zededa maybe I'm arguing we should remove it from the attestation API. Thoughts?

Reply:
If location-based attestation has not been required, then it is probably safer to not support it and possibly remove the APIs. Otherwise we would need to experiment to evaluate how reliable and practical it would be, but most likely it will only introduce false negatives (i.e. attestation failed even if it should not).

Reply:
Right now I put it here to have a possibility to see what device send during attestation. Location lock is defined inside the documentation https://wiki.lfedge.org/plugins/servlet/mobile?contentId=27722830#content/view/27722830, but correct, I did not define any checks.

Reply:
Removed for now